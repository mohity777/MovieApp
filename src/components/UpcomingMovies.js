import React from 'react';
import { useSelector } from 'react-redux';
import { View, Text, StyleSheet, FlatList } from 'react-native';
import MovieCard from './MovieCard';

const UpcomingMovies = props => {

    const { upComingMovies, upcomingLoading } = useSelector(state => state.movies);

    return(
       <View>
           <Text style={styles.headline}>Upcoming Movies</Text>
           <FlatList 
             data={upcomingLoading ? [1,2,3,4] : upComingMovies}
             keyExtractor={(_,i) => i.toString()}
             renderItem={({item, index}) => <MovieCard loading={upcomingLoading} index={index} item={item}/>}
             horizontal={true}
             showsHorizontalScrollIndicator={false}
           />
       </View>
    )
}

export default UpcomingMovies;

const styles = StyleSheet.create({
  headline: {
      fontFamily: 'Montserrat-Bold',
      color: 'white',
      fontSize: 15,
      marginBottom: 30,
  }
})