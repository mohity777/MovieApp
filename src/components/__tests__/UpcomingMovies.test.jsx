import React from 'react';
import {create, act} from 'react-test-renderer';
import UpcomingMovies from '../UpcomingMovies';
import * as Redux from 'react-redux';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('UpcomingMovies test cases', () => {
  const state = {
    movies: {
      upComingMovies: [],
      upcomingLoading: false,
    },
  };

  beforeEach(() => {
    Redux.useSelector.mockImplementation(callback => {
      return callback(state);
    });
  });

  it('renders correctly', () => {
    let root;
    act(() => {
      root = create(<UpcomingMovies />);
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
