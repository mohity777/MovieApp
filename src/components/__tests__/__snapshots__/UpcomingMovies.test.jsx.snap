// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingMovies test cases renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 15,
        "marginBottom": 30,
      }
    }
  >
    Upcoming Movies
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
